### ISSUE TITLE
Whitelisting IPs using traefik or ufw

### ISSUE THREAD

**Issue by HannanSolo** [eth-educators/eth-docker#1247]
I am running my execution client RPC through traefik as I want it public but to only allow a whitelist of IP's to access it. 

Is there a recommended way of adding a whitelist to traefik or putting UFW on the server with traefik?

Thank you.

**Comment by yorickdowne** [comment]
Yes, there is. You can place ufw "in front of" docker, instructions here: https://eth-docker.net/Support/Cloud

**Comment by HannanSolo** [comment]
The tutorial only states to use it if not encrypting the traffic with traefik. What is the recommended setup while using Traefik?

If I want to whitlist IPs then is it better to unencrypt the traffic and expose the ports while having ufw?

**Comment by yorickdowne** [comment]
You can use it for traefik just as well. 443 in that case. It’s a more specific use case and one most users don’t have; you can absolutely do things that way.

**Comment by HannanSolo** [comment]
I followed the tutorial and debugged a but with no success.

I installed and ran ufw denying all incoming and allowing all outgoing traffic.

This is my rules.
30303                      ALLOW       Anywhere                  
OpenSSH                    ALLOW       Anywhere                  
443/tcp                    DENY        Anywhere                  
443                        DENY        Anywhere                  
30303 (v6)                 ALLOW       Anywhere (v6)             
OpenSSH (v6)               ALLOW       Anywhere (v6)             
443/tcp (v6)               DENY        Anywhere (v6)             
443 (v6)                   DENY        Anywhere (v6)   

Yet, I able to run curl commands to my RPC layer.

It seems how ethd is setup restricts ufw from touching it.

**Comment by yorickdowne** [comment]
As long as you followed the instructions on placing ufw "in front of" docker, you should not be able to curl to this from the Internet / a public IP.

**Comment by HannanSolo** [comment]
I updated the after.rules and before.init, and I am able to block ports only when specified.  One step in the right direction :)

Keeping the 443 DENY Anywhere blocks the RPC communication, but when I do not have any rules on 443 and use `sudo ufw default deny incoming` I am still able to curl.

This time I made sure to read through the documentation and cannot find what I am missing.

Thanks again for your help.

**Comment by yorickdowne** [comment]
Yes this is deliberate. The default deny doesn’t impact docker in this setup. The idea is that even with ufw “in front of”, stuff like P2P ports still works by default. All mapped docker ports remain reachable by default. What’s in front of is the user table, not all of ufw.

the user can then decide to close off ports, for example
allow proto tcp from <trusted-ip> to any port 443 comment “RPC for grandma”
deny proto tcp from any to any port 443

**Comment by HannanSolo** [comment]
That worked! Today I learned that the order of ufw rules matters a lot.

I thought a deny rule had priority over all others.

Thank you for your help.

**Comment by yorickdowne** [comment]
Glad that worked! 

Yes ufw works like every other stateful firewall in the market: Rules are evaluated in order, top down.
